src/main/java/org/jitsi/jicofo/bridge/BridgeSelectionStrategy.java

```diff
@@ -86,63 +139,215 @@ public Bridge select(
+     * @param participantRegion the participant region.
+     *
+     * @return an optional that contains the least loaded bridge in the
+     * participant's region if it exists.
+     */
+    Optional<Bridge> leastLoadedInRegion(
+            List<Bridge> bridges,
+            Map<Bridge, Integer> conferenceBridges,
+            String participantRegion)
+    {
+        Optional<Bridge> result = bridges.stream()
+            .filter(inRegion(participantRegion))
+            .findFirst();
+
+        if (result.isPresent())
         {
-            bridge = findFirstOperationalInRegion(bridges, localRegion);
+
+            logger.info("bridge_selected,bridge="  + result.get() +
+                "participant_region " + participantRegion);
+
+            totalLeastLoadedInRegion++;
+
+            updateSplitStats(conferenceBridges, result.get(), participantRegion);
         }
 
-        // Otherwise, just find the first operational bridge.
-        if (bridge == null)
+        return result;
+    }
+
+    /**
+     * Finds the least loaded non overloaded bridge that is already handling the conference.
+     *
+     * @param bridges the list of operational bridges, ordered by load.
+     * @param conferenceBridges the set of bridges that are already used in the
+     * conference.
+     *
+     * @return an optional that contains the least loaded bridge that is already
+     * in the  conference, if it exists.
+     */
+    Optional<Bridge> nonLoadedAlreadyInConference(
+            List<Bridge> bridges,
+            Map<Bridge, Integer> conferenceBridges,
+            String participantRegion)
+    {
+        Optional<Bridge> result = bridges.stream()
+            .filter(not(b -> isOverloaded(b, conferenceBridges)))
+            .filter(selectFrom(conferenceBridges.keySet()))
+            .findFirst();
+
+        if (result.isPresent())
         {
-            bridge = findFirstOperationalInRegion(bridges, null);
+            logger.info("bridge_selected,bridge=" + result.get() +
+                ",conference_bridges=" + conferenceBridges.keySet().stream().map(Bridge::toString).collect(Collectors.joining(", ")) +
+                "participant_region " + participantRegion);
+            totalLeastLoadedAlreadyInConference++;
         }
 
-        return bridge;
+        return result;
     }
 
     /**
-     * Returns the first operational bridge in the given list which matches
-     * the given region (if the given regio is {@code null} the region is
-     * not matched).
+     * Finds the least loaded bridge.
+     *
+     * @param bridges the list of operational bridges, ordered by load.
      *
-     * @param bridges
-     * @param region
-     * @return
+     * @return an optional that contains the least loaded bridge if it exists.
      */
-    private Bridge findFirstOperationalInRegion(
+    Optional<Bridge> leastLoaded(
             List<Bridge> bridges,
-            String region)
+            Map<Bridge, Integer> conferenceBridges,
```

Comment:
It's a bummer we have to take in `conferenceBridges` and `participantRegion` here just to call `updateSplitStats`...is there another place we could call `updateSplitStats` from? Or another way to track that?

Reply:
It's also for consistency -- all these methods in the base class that make a selection *and* update the stats (i.e. assume their selection was used) have the same prototype. 

Since calling `updateSplitStats` only happens in some cases, we would have to move this logic to the `doSelect` implementations which would be quite awkward and confusing:
https://github.com/jitsi/jicofo/pull/428/files#diff-d6bf891e4163817e6f64797ce48fbb1aR47
https://github.com/jitsi/jicofo/pull/428/files#diff-9c48bdfd8bc771eb097b1aa7f372a011R69

Reply:
Ok, I had commented to george before that I thought it was misleading to pass values (e.g. `conferenceBridges` and `participantRegion`) to methods that don't actually take them into account.